src/linker/configuration.py

```diff
@@ -140,10 +154,30 @@ def _load_input_data_paths(input_data: Path) -> List[Path]:
         return file_list
 
     @staticmethod
-    def _get_spark_requests(
-        environment: Dict[str, Union[Dict, str]]
-    ) -> Optional[Dict[str, Any]]:
-        spark = environment.get("spark", None)
-        if spark and not "keep_alive" in spark:
-            spark["keep_alive"] = False
-        return spark
+    def _assign_default_environment(
+        subenv: Dict[str, Any], defaults: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Recursively fill in missing values with defaults."""
+        required_keys = ["computing_environment", "container_engine"]
```

Comment:
Should this live in metadata or something?

Reply:
I'd prefer to keep it in this module, but good point, I'll create the list up top